plugins/node/opentelemetry-instrumentation-express/src/types.ts

```diff
@@ -88,6 +88,20 @@ export type SpanNameHook = (
   defaultName: string
 ) => string;
 
+/**
+ * Function that can be used to add custom attributes to the current span or the root span on
+ * a Express request
+ * @param requestSpan - The Express request handler span
+ * @param expressRequest: - The Express request object
+ */
+export interface ExpressRequestCustomAttributeFunction {
+  (
+    span: Span,
+    layerType: ExpressLayerType,
+    expressRequest: PatchedRequest
```

Comment:
please store these properties in an info object, so that if more data is added in the future, we will not end up with huge function signatures.

You can see an example in this file for the spanNameHook which stores all the relevant data in `ExpressRequestInfo` (or even reuse it?)